core/src/main/java/feign/kotlinSupport/ClassUtils.java

```diff
@@ -0,0 +1,423 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feign.kotlinSupport;
+
+import java.io.Closeable;
+import java.io.Externalizable;
+import java.io.Serializable;
+import java.lang.reflect.*;
+import java.util.*;
+
+/**
+ * Miscellaneous {@code java.lang.Class} utility methods.
+ * Mainly for internal use within the framework.
+ *
+ * @author Juergen Hoeller
+ * @author Keith Donald
+ * @author Rob Harrop
+ * @author Sam Brannen
+ * @since 1.1
+ */
+public abstract class ClassUtils {
```

Comment:
This class (and others on the PR) look like copy and paste from somewhere.... please use the original library

Reply:
https://github.com/OpenFeign/feign/pull/1706#issuecomment-1202975619

> Assert.java / ClasUtils.java / KotlinDetector.java is from spring core project -> I will refactor when the implementation approach is agreed upon.

If the approach is agreed upon, Assert.java / ClasUtils.java used in KotlinDetector and KotlinDetector implementation will be refactored. These classes are from [Spring Core](https://github.com/spring-projects/spring-framework/blob/main/spring-core/src/main/java/org/springframework/util/ClassUtils.java). If I use the original library, I must use the Spring Core library.

How about reviewing where the feign code branches off using KotlinDetector for Kotlin support? I wrote the code only to demonstrate the approach.

Reply:
Assert.java -> Removed
ClasUtils.java / KotlinDetector.java -> Refactored